dvc/fs/azure.py

```diff
@@ -69,8 +69,8 @@ def _prepare_credentials(self, config):
         login_info["client_id"] = config.get("client_id")
         login_info["client_secret"] = config.get("client_secret")
 
-        if not any(login_info.values()):
-            login_info["credentials"] = DefaultAzureCredential()
+        if not any([a in login_info and a is not None for a in ["connection_string", "account_key", "sas_token", "client_secret"]]):
```

Comment:
```py
        if not any(
            [
                a in login_info and a is not None
                for a in [
                    "connection_string",
                    "account_key",
                    "sas_token",
                    "client_secret",
                ]
            ]
        ):
```

This condition seems like always succeeding. As you can see `a` is always a string from that list which is non none, and `login_info` always contains it (even though if you don't specify it in the config)

Reply:
Ah sorry should be `login_info[a] is not None`

Reply:
Can you explain why you excluded other values from here (e.g `tenant_id`). AFAIK `DefaultCredentials` will override the specified `tenant_id` in your config so I am not so sure that would be wanted behavior.,

Reply:
Just that all three of tenant_id, client_id and client_secret are required for SPA so it seemed superfluous to check against all